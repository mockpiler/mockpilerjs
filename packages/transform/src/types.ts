export type TransformFn<V = any, R = any> = (value: V) => R
