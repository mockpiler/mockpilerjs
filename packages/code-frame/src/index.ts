/**
 * Expose buffer-related stuff
 */
export * from './buffer'

/**
 * Expose code-frame main class
 */
export * from './code-frame'

/**
 * Expose options
 */
export * from './options'
