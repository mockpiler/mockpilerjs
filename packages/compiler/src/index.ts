/**
 * Expose compiler main functionalities
 */
export * from './compiler'

/**
 * Expose types
 */
export * from './types'

/**
 * Expose alias
 */
export { createCompiler as mock } from './compiler'

/**
 * Expose context utils
 */
export * from '@mockpiler/context'

/**
 * Expose transformers
 */
export * from '@mockpiler/transform'
