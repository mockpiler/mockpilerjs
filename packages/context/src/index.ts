/**
 * Expose context-related
 */
export * from './context'

/**
 * Expose utils
 */
export * from './utils'

/**
 * Expose context types
 */
export * from './types'
