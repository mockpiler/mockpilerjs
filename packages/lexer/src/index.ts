/**
 * Expose lexer-related things
 */
export * from './lexer'

/**
 * Expose useful constants
 */
export * from './constants'

/**
 * Expose token declarations
 */
export * from './token'
