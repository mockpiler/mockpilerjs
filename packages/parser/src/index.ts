/**
 * Expose AST nodes
 */
export * from './ast'

/**
 * Expose parser-related things
 */
export * from './parser'
