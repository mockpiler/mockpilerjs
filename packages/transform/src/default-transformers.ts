import type { TransformFn } from './types'

const MapTransform: TransformFn<
  Array<[unknown, unknown]> | Record<string, unknown>,
  Map<any, any>
> = value => {
  if (Array.isArray(value)) {
    return new Map(value)
  }

  if (typeof value !== 'object' || value === null) {
    throw new TypeError('Invalid value')
  }

  const map = new Map()
  const keys = Object.keys(value)

  for (const key of keys) {
    map.set(key, value[key])
  }

  return map
}

const SetTransform: TransformFn<any[], Set<any>> = value => {
  if (!Array.isArray(value)) {
    throw new TypeError('Invalid value')
  }

  return new Set(value)
}

export const transformers = {
  ToMap: MapTransform,
  ToSet: SetTransform
}
