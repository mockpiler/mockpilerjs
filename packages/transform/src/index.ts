/**
 * Expose default transform functions
 */
export * from './default-transformers'

/**
 * Expose types
 */
export * from './types'
