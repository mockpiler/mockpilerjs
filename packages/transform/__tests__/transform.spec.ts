import { transformers } from '../src'

describe('transform', () => {
  test('should apply ToMap transform correctly', () => {
    expect(
      transformers.ToMap([
        ['key', 'value'],
        [{}, []]
      ])
    ).toMatchSnapshot()

    expect(
      transformers.ToMap({
        name: 'John Doe',
        age: 33
      })
    ).toMatchSnapshot()
  })

  test('should apply ToSet transform correctly', () => {
    expect(
      transformers.ToSet(['name', 'John Doe', 'age', 33])
    ).toMatchSnapshot()
  })
})
