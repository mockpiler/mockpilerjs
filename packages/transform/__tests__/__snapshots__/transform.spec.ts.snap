// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should apply ToMap transform correctly 1`] = `
Map {
  "key" => "value",
  Object {} => Array [],
}
`;

exports[`transform should apply ToMap transform correctly 2`] = `
Map {
  "name" => "John Doe",
  "age" => 33,
}
`;

exports[`transform should apply ToSet transform correctly 1`] = `
Set {
  "name",
  "John Doe",
  "age",
  33,
}
`;
